modules/post/windows/gather/enum_av_excluded.rb

```diff
@@ -0,0 +1,148 @@
+        OptBool.new('ESSENTIALS', [true, 'Enumerate exclusions for Microsoft Security Essentials', true]),
+        OptBool.new('SEP', [true, 'Enumerate exclusions for Symantec Endpoint Protection (SEP)', true])
+      ]
+    )
+  end
+
+  def enum_mssec
+    if registry_enumkeys("HKLM\\SOFTWARE\\Microsoft").include?("Microsoft Antimalware")
+      print_status "MS Security Essentials Identified"
+      return true
+    else
+      return false
+    end
+  rescue
+    return false
+  end
+
+  def enum_defender
+    if registry_enumkeys("HKLM\\SOFTWARE\\Microsoft").include?("Windows Defender")
+      print_status "Windows Defender Identified"
+      return true
+    else
+      return false
+    end
+  rescue
+    return false
+  end
+
+  def enum_sep
+    if registry_enumkeys("HKLM\\SOFTWARE\\Symantec").include?("Symantec Endpoint Protection")
+      print_status "SEP Identified"
+      return true
+    else
+      return false
+    end
+  rescue
+    return false
+  end
+
+  def excluded_sep
+    print_status "Excluded Locations:"
+    keyadm = "HKLM\\SOFTWARE\\Symantec\\Symantec Endpoint Protection\\AV\\Exclusions\\ScanningEngines\\Directory\\Admin"
+    if (found_keysadm = registry_enumkeys(keyadm))
+      found_keysadm.each do |vals|
+        full = keyadm + "\\" + vals
+        values = registry_getvaldata(full, "DirectoryName")
+        print_good "#{values}"
+      end
+    else
+      print_error "No Admin Locations Found"
+    end
+
+    keycli = "HKLM\\SOFTWARE\\Symantec\\Symantec Endpoint Protection\\AV\\Exclusions\\ScanningEngines\\Directory\\Client"
+    if (found_keyscli = registry_enumkeys(keycli))
+      found_keyscli.each do |vals|
+        full = keycli + "\\" + vals
+        values = registry_getvaldata(full, "DirectoryName")
+        print_good "#{values}"
+      end
+    else
+      print_error "No Client Locations Found"
+    end
+  end
+
+  def excluded_mssec
+    print_status "Excluded Locations:"
+    keyms = "HKLM\\SOFTWARE\\Microsoft\\Microsoft Antimalware\\Exclusions\\Paths\\"
+    if (found = registry_enumvals(keyms))
+      found.each do |num|
+        print_good "#{num}"
+      end
+    else
+      print_error "No Excluded Locations Found"
+    end
+  end
+
+  def excluded_defender
+    print_status "Excluded Locations:"
+    keyms = "HKLM\\SOFTWARE\\Microsoft\\Windows Defender\\Exclusions\\Paths\\"
+    if (found = registry_enumvals(keyms))
+      found.each do |num|
+        print_good "#{num}"
+      end
+    else
+      print_error "No Excluded Locations Found"
+    end
+  end
+
+  def setup
+    unless datastore['DEFENDER'] || datastore['ESSENTIALS'] || datastore['SEP']
+      fail_with(Failure::BadConfig, 'Must set one or more of DEFENDER, ESSENTIALS or SEP to true')
+    end
+  end
+
+  def run
+    if sysinfo['Architecture'] =~ /WOW64/
+      print_error "You are running this module from a 32-bit process on a 64-bit machine. Migrate to a 64-bit process and try again"
```

Comment:
I just tested this against two nearly identical Windows 7 SP1 64-bit hosts and a 2008 R2 SP1 with defender installed and configured with some exclusions, and only one of the hosts was able to find exclusions without migrating to a 64-bit process, but that was because for whatever reason, the meterpreter session I got on the two Windows 7 hosts were different at least according to what was in `sysinfo`:

```
 {"Computer"=>"ESSENTIALS-11-P",
 "OS"=>"Windows 7 (Build 7601, Service Pack 1).",
 "Architecture"=>"x64",
 "System Language"=>"en_US",
 "Domain"=>"WORKGROUP",
 "Logged On Users"=>1}
{"Computer"=>"ESSENTIALS-12-P",
 "OS"=>"Windows 7 (Build 7601, Service Pack 1).",
 "Architecture"=>"x64 (Current Process is WOW64)",
 "System Language"=>"en_US",
 "Domain"=>"WORKGROUP",
 "Logged On Users"=>1}
```

So... that wonky possibly new and unrelated defect aside, this means that the code as currently written will not be able to enumerate these exclusions if the process from which it is launched is the same architecture as the target.  

I swear this was fixed already, so I dug.  As it turns out, *I* filed the ticket (https://github.com/rapid7/metasploit-framework/issues/4589) and @bcook-r7 fixed it in https://github.com/rapid7/metasploit-framework/pull/4614.  Let me try to use this new code here and ensure that it works.